src/Manager.php

```diff
@@ -40,7 +40,7 @@ public function getSearchMeta(object $object): array
         $ids = [];
 
         foreach ($this->handlers as $handler) {
-            if ($handler->handles($class) === false) {
+            if ($handler->getHandledClass() !== $class) {
```

Comment:
As discussed previously, we have search handlers that handle multiple classes - so this needs to be changed that they can return an array of classes to be supported.

I would also instead of doing a straight !== class check, do a class_implements check.